dashboard/app/assets/javascripts/header.js

```diff
@@ -445,16 +445,32 @@ function mergedActivityCssClass(a, b) {
 }
 
 function populateProgress(scriptName) {
-  var scriptProgress = dashboard.clientState.allLevelsProgress()[scriptName] || {};
+  var clientProgress = dashboard.clientState.allLevelsProgress()[scriptName] || {};
+  $.ajax('/api/user_progress/' + scriptName).done(function (data) {
```

Comment:
We should a handle for the case when the Ajax request fails or times out that just uses the client progress, otherwise the user ends up with no progress (?).

If we could possibly show the client progress initially and then add the server progress when it comes in, that would be ideal.